5.0/en/0x51-V51-OAuth2.md

```diff
@@ -28,6 +75,13 @@ Given the complexity of the area, it is vitally important for a secure OAuth or
 
 ## V51.3 OAuth Resource Server
 
+In the context of ASVS and this chapter, the resource server is an API. To provide secure access the resource server must:
+
+* Validate the access token, according to token format and protocol specifications
+* If valid, enforce authorization decisions based on information from the access token and granted permissions. For example, the resource server needs to verify that the client (acting on behalf of RO) is authorized to access the requested resource.
+
+Thus the requirements listed here are OAuth or OIDC specific and should be performed after token integrity validation and before accepting the delegated authorization information from the token.
```

Comment:
I find "and before accepting the delegated authorization information from the token" a little hard to understand.
To make this clearer, perhaps change to something like this "and before performing authorization based on information from the token"

Reply:
Nitpick about:

> Thus the requirements listed here are OAuth or OIDC specific and should be performed after token integrity validation 

The requirements are to "the verifier must verify that …" therefore they cannot really be performed after "token integrity validation".

Reply:
> I am wondering if what is being expressed in this sentence should not be a requirement instead?

Rereading these requirements, I find it strange (somewhat lacking) that there is no requirement *in this section* requiring the verification of the token's authenticity (same for OIDC ID tokens in the OIDC RP section).

It is implied by "2.11.2" (for the self-encoded cases, but not for the "token-introspection" case) but I am wondering if this is still somehow lacking here.

Strawman proposition of a requirement:

> Verify the resource server, checks the validity of the access token before using the associated claims. This is usually done by verification of a digital signature or through token introspection.

Reply:
Please open a separate issue for that

Reply:
How the proposed requirement could be not duplicate of 52.1.1?

Reply:
It overlaps a lot, but it includes "or through token introspection". Maybe add something like this (as a separate issue (less overlap?)

> Verify the resource server, checks the validity of the access token before using the associated claims. For reference tokens this is done by through token introspection.

Reply:
> How the proposed requirement could be not duplicate of 52.1.1?

Yes, the token introspection is note is current 52.1.1 and not anywhere else?

Reply:
After re-reading the different existing requirements, my proposition might not be necessary after all.

Reply:
Maybe it would be good to mention introspection in the chapter text instead? 

> In the context of ASVS and this chapter, the resource server is an API. To provide secure access the resource server must:
>
>* Validate the access token, according to token format and protocol specifications, **e g JWT-validation or OAuth token introspection.**
>* If valid, enforce authorization decisions based on information from the access token and granted permissions. For >example, the resource server needs to verify that the client (acting on behalf of RO) is authorized to access the >requested resource.

>Thus the requirements listed here are OAuth or OIDC specific and should be performed after token validation **and before performing authorization based on information from the token**.